modules/data_warehouse/main.tf

```diff
@@ -0,0 +1,396 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+data "google_project" "project" {
+  project_id = var.project_id
+}
+
+module "project-services" {
+  source                      = "terraform-google-modules/project-factory/google//modules/project_services"
+  version                     = "13.0.0"
+  disable_services_on_destroy = false
+
+  project_id  = var.project_id
+  enable_apis = var.enable_apis
+
+  activate_apis = [
+    "compute.googleapis.com",
+    "cloudapis.googleapis.com",
+    "cloudbuild.googleapis.com",
+    "datacatalog.googleapis.com",
+    "datalineage.googleapis.com",
+    "eventarc.googleapis.com",
+    "bigquerymigration.googleapis.com",
+    "bigquerystorage.googleapis.com",
+    "bigqueryconnection.googleapis.com",
+    "bigqueryreservation.googleapis.com",
+    "bigquery.googleapis.com",
+    "storage.googleapis.com",
+    "storage-api.googleapis.com",
+    "run.googleapis.com",
+    "pubsub.googleapis.com",
+    "bigqueryconnection.googleapis.com",
+    "cloudfunctions.googleapis.com",
+    "bigquerydatatransfer.googleapis.com",
+    "artifactregistry.googleapis.com",
+  ]
+}
+
+#random id
+resource "random_id" "id" {
+  byte_length = 4
+}
+
+# Set up service account for the Cloud Function to execute as
+resource "google_service_account" "cloud_function_service_account" {
+  project      = var.project_id
+  account_id   = "cloud-function-sa-${random_id.id.hex}"
+  display_name = "Service Account for Cloud Function Execution"
+}
+
+# TODO: scope this down
+resource "google_project_iam_member" "cloud_function_service_account_editor_role" {
+  project = var.project_id
+  role    = "roles/editor"
+  member  = "serviceAccount:${google_service_account.cloud_function_service_account.email}"
+
+  depends_on = [
+    google_service_account.cloud_function_service_account
+  ]
+}
+
+# TODO: Add scheduled load job
+
+# Set up Storage Buckets
+# # Set up the raw storage bucket
+resource "google_storage_bucket" "raw_bucket" {
+  name                        = "ds-edw-raw-${random_id.id.hex}"
+  project                     = var.project_id
+  location                    = var.region
+  uniform_bucket_level_access = true
+  force_destroy               = true
```

Comment:
Let's expose this as a var with default false so user's won't accidentally destroy

Reply:
done